### PR TITLE
Adds functionality for tracking and logging avg and best loss during training

### PR DESCRIPTION
**Pull request by geoffroeder**
To monitor convergence when training deep learning models, it is helpful to have a log of the best validation or training loss seen so far, as well as the average loss over the all epochs so far.

This PR adds that functionality into `ignite.trainer.Trainer`.

### CODE REVIEW

**Comment by alykhantejani** [comment]
Hey @geoffroeder, as discussed offline you can now achieve this functionality through `ignite.handlers.logging` or using the `simple_moving_average` etc. in the `History` object.
